GPU plugin: add xpumd 2.x as alternative health data source (#2279)
intel/intel-device-plugins-for-kubernetes: cmd/gpu_plugin/README.md

| -enable-monitoring | - | disabled | Enable '*_monitoring' resource that provides access to all Intel GPU devices on the node, [see use](./monitoring.md) |
| -monitoring-mode | string | single | How monitoring resources are registered: single or split |
| -health-management | - | disabled | Enable health management by requesting data from oneAPI/Level-Zero interface. Requires [GPU Level-Zero](../gpu_levelzero/) sidecar. See [health management](#health-management) |

> | -xpumd-endpoint | string | "" | Unix socket path for xpumd health service (e.g. `/run/xpumd/intelxpuinfo.sock`). When set, xpumd is used as the health data source instead of the Level-Zero sidecar. Cannot be combined with `-health-management` or temperature limit flags. See [xpumd health source](#xpumd-health-source) |

Reviewer: ```suggestion
| -xpumd-endpoint | string | "" | Unix socket path for xpumd health service (e.g. `/run/xpumd/intelxpuinfo.sock`). When set, xpumd is used as the health data source instead of the Level-Zero sidecar. Cannot be combined with `-health-management`. Temperature limits are specified in xpumd service configuration, instead of GPU plugin flags. See [xpumd health source](#xpumd-health-source) |
```

> | -xpumd-endpoint | string | "" | Unix socket path for xpumd health service (e.g. `/run/xpumd/intelxpuinfo.sock`). When set, xpumd is used as the health data source instead of the Level-Zero sidecar. Cannot be combined with `-health-management`. Temperature limits are specified in xpumd service configuration, not with GPU plugin flags. See [xpumd health source](#xpumd-health-source) |

Reviewer 2: the gpudeviceplugin controller part comes in a later PR?

Reviewer 3: Possible, but not likely. I'm prioritizing integration with the gpu operator.

> | -xpumd-endpoint | string | "" | Unix socket path for xpumd health service (e.g. `/run/xpumd/intelxpuinfo.sock`). When set, xpumd is used as the health data source instead of the Level-Zero sidecar. Cannot be combined with `-health-management`. Temperature limits are specified in xpumd service configuration, instead of GPU plugin flags. See [xpumd health source](#xpumd-health-source) |

Reviewer: Marginally better wording than my earlier suggestion:
```suggestion
| -xpumd-endpoint | string | "" | Unix socket path for xpumd health service (e.g. `/run/xpumd/intelxpuinfo.sock`). When set, xpumd is used as the health data source instead of the Level-Zero sidecar. Cannot be combined with `-health-management`. Temperature limits are specified in xpumd service configuration, not with GPU plugin flags. See [xpumd health source](#xpumd-health-source) |
```

Temperature limit can be provided via the command line argument, default is 100C.

### xpumd health source

> As an alternative to the Level-Zero sidecar, GPU plugin can obtain device health data from [Intel XPU Manager (xpumd)](https://github.com/intel/xpumanager) 2.x. xpumd exposes a local gRPC streaming API (`WatchDeviceHealth`) that provides equivalent health information without requiring a privileged sidecar container.

Reviewer: Drop technical detail and concentrate on usage:
```suggestion
As an alternative to the Level-Zero sidecar, GPU plugin can obtain device health data from [Intel XPU Manager (xpumd)](https://github.com/intel/xpumanager) v2.x that provides equivalent health information without requiring a privileged sidecar container.
```

When xpumd is running on the host it creates a unix socket (default: `/run/xpumd/intelxpuinfo.sock`). The GPU plugin connects to this socket and streams health events. A device is reported as `Unhealthy` if any health domain reports severity `WARNING` or higher.

To use xpumd as the health source, deploy the plugin with the provided Kustomize overlay:

```bash
kubectl apply -k 'https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/gpu_plugin/overlays/xpumd?ref=<RELEASE_VERSION>'
```

> The overlay mounts `/run/xpumd` from the host (read-only) into the plugin pod and passes the `-xpumd-endpoint` and `-enable-monitoring` flags automatically.

Reviewer: ```suggestion
The overlay mounts `/run/xpumd` from the host (read-only) into the plugin pod and passes the required `-xpumd-endpoint` and `-enable-monitoring` flags automatically.
```

> > **Note**: `-xpumd-endpoint` and `-health-management` are mutually exclusive. Temperature limit flags (`-temp-limit`, `-gpu-temp-limit`, `-memory-temp-limit`) are not applicable when using the xpumd health source.

Reviewer: ```suggestion
> **Note**: `-xpumd-endpoint` and (sidecar) `-health-management` flags are mutually exclusive. Sidecar specific temperature limit flags (`-temp-limit`, `-gpu-temp-limit`, `-memory-temp-limit`) are not applicable when using `xpumd` as health source.
```
